code/modules/antagonists/heretic/knowledge/cosmic_lore.dm

```diff
@@ -243,10 +250,28 @@
 	. = ..()
 	priority_announce("[generate_heretic_text()] A Star Gazer has arrived into the station, [user.real_name] has ascended! This station is the domain of the Cosmos! [generate_heretic_text()]","[generate_heretic_text()]", ANNOUNCER_SPANOMALIES)
 	var/mob/living/basic/star_gazer/star_gazer_mob = new /mob/living/basic/star_gazer(loc)
+	star_gazer_mob.maxHealth = INFINITY
+	star_gazer_mob.health = INFINITY
+	user.AddElement(/datum/element/death_linked, star_gazer_mob)
 	star_gazer_mob.AddComponent(/datum/component/obeys_commands, star_gazer_commands)
+	star_gazer_mob.AddComponent(/datum/component/damage_aura, range = 7, burn_damage = 0.5, simple_damage = 0.5, has_owner = user)
 	star_gazer_mob.befriend(user)
+	var/datum/action/cooldown/open_mob_commands/commands_action = new /datum/action/cooldown/open_mob_commands()
+	commands_action.Grant(user, star_gazer_mob)
 	var/datum/action/cooldown/spell/touch/star_touch/star_touch_spell = locate() in user.actions
 	if(star_touch_spell)
 		star_touch_spell.set_star_gazer(star_gazer_mob)
+		star_touch_spell.ascended = TRUE
+
+	var/datum/antagonist/heretic/heretic_datum = IS_HERETIC(user)
```

Comment:
This works but don't do it, it's ugly 😔 
While very unlikely it's always possible that some day someone will refactor this macro to return a boolean like it implies that it does and then your code will break.
```suggestion
	var/datum/antagonist/heretic/heretic_datum = user.mind.has_antag_datum(/datum/antagonist/heretic)
```